neuralmonkey/encoders/cnn_encoder.py

```diff
@@ -202,28 +204,30 @@ def rnn_encoder(inputs, last_layer_size, scope):
             encoder_state = rnn_encoder(
                 encoder_ins, last_layer_size, "encoder-forward")
 
+            # pylint: disable=redefined-variable-type
             if bidirectional:
                 backward_encoder_state = rnn_encoder(
                     list(reversed(encoder_ins)),
-                    last_layer_size,
-                    "encoder-backward")
-                # pylint: disable=redefined-variable-type
+                    last_layer_size, "encoder-backward")
                 encoder_state = tf.concat(
                     1, [encoder_state, backward_encoder_state])
 
             self.encoded = encoder_state
 
-            self.attention_tensor = \
+            self.__attention_tensor = \
                 tf.reshape(last_layer, [-1, image_width,
                                         last_n_channels * image_height])
 
-            att_in_weights = tf.squeeze(
+            self.__attention_weights = tf.squeeze(
                 tf.reduce_prod(last_padding_masks, [1]), [2])
 
-            self.attention_object = Attention(self.attention_tensor,
-                                              scope="attention_{}".format(
-                                                  name),
-                                              input_weights=att_in_weights)
+    @property
+    def _attention_tensor(self):
+        return self.__attention_tensor
+
+    @property
+    def _attention_mask(self):
+        return self.__attention_weights
```

Comment:
Z dvěma podržítkama je to private field, s jedním je to property odděděná od attentive.